### ISSUE TITLE
@rid values cannot be compared with !=, unlike all other scalar types

### ISSUE THREAD

**Issue by obi1kenobi** [orientechnologies/orientdb#6605]
## OrientDB Version, operating system, or hardware.
- v2.0 SNAPSHOT[ ] - .18[ ] .17[ ] .16[ ] .15[ ] .14[ ] .13[ ] .12[ ] .11[ ] .10[ ] .9[ ] .8[ ] .7[ ] .6[ ] .5[ ] .4[ ] .3[ ] .2[ ] .1[ ] .0[ ]
- v2.1 SNAPSHOT[ ] - .16[ ] .15[ ] .14[ ] .13[ ] .12[ ] .11[ ] .10[ ] .9[ ] .8[ ] .7[ ] .6[ ] .5[ ] .4[ ] .3[ ] .2[ ] .1[ ] .0[ ]
- v2.2 SNAPSHOT[ ] - .7 [x] .rc1[ ] .beta2[ ] .beta1[ ]

## Operating System
- [x] Linux
- [x] MacOSX
- [ ] Windows
- [ ] Other Unix
- [ ] Other, name? 

## Expected behavior and actual behavior

Expected: @rid values can be checked for inequality with both `!=` and `<>`, similarly to other values like strings and numbers.

Actual: @rid values can only be checked for inequality with `<>`. Trying to use `!=` raises the following error:
```
java.lang.ClassCastException: com.orientechnologies.orient.code.id.ORecordId cannot be cast to java.lang.Boolean
```

Worse still, in combination with #6604 this means that entire columns from the output are silently dropped for no obvious reason, as #6604 will swallow the raised error in the `eval` statement.

## Steps to reproduce the problem

Follow the setup from #6604, then run:
```
select @rid, name, if(eval("@rid != @rid"), "match", "no match") as m from V1 where @rid <> #12:0
```

Observe the error I listed above. Then, remove the `WHERE` clause, and observe that the returned output is missing the `m` column, as in #6604.

**Comment by lvca** [comment]
Have you tried with `<>` instead of `!=`?

**Comment by obi1kenobi** [comment]
As I mentioned above, `<>` works fine both for rids and other types. However, all other types that I tested (strings, numbers etc.) also accept `!=` to check inequality, so the asymmetry is unexpected and confusing.

**Comment by lvca** [comment]
@luigidellaquila it seems an error on the old parser.

**Comment by luigidellaquila** [comment]
Hi @obi1kenobi @lvca 

I'm checking it

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi @obi1kenobi 

I just fixed it, the fix is in 2.2.x branch and will be released in 2.2.11

Thanks

Luigi

**Comment by obi1kenobi** [comment]
Thanks!
